engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala

```diff
@@ -63,70 +63,66 @@ class UpsertVisualizationJob(
     ctx.locking.withContextLock(
       config.executionContextId,
       this.getClass,
-      () =>
-        ctx.locking.withWriteCompilationLock(
-          this.getClass,
-          () => {
-            val maybeCallable =
-              UpsertVisualizationJob.evaluateVisualizationExpression(
-                config.visualizationModule,
-                config.expression
-              )
-
-            maybeCallable match {
-              case Left(ModuleNotFound(moduleName)) =>
-                ctx.endpoint.sendToClient(
-                  Api.Response(Api.ModuleNotFound(moduleName))
-                )
-                None
+      () => {
```

Comment:
How are the changes in this source related to module scope builder?

Reply:
We can now reduce the scope of the lock, which was previously experiencing a race condition because of mutable `ModuleScope`.